### PR TITLE
docs: clarify data type in Forms and Mutations section

### PR DESCRIPTION
**Pull request by michaelangeloio**
### What?

Clarify the "Forms and Mutations" docs. 

### Why?

This may be opinionated, but I believe this section of the docs starts out assuming that the user knows how the `<form>` HTML works. Coming from the client side React world, most developers are used to using state in forms. Clarifying that this data is accessible via the standard web API [FormData](https://developer.mozilla.org/en-US/docs/Web/API/FormData/FormData) is essential for both transitioning developers and new developers.

### CODE REVIEW

**Review by leerob — DISMISSED** [review]

**Comment by ijjk** [comment]
Allow CI Workflow Run

- [ ] approve CI run for commit: a41ef89c619de929654d5126da76c79a67a5d6dd

Note: this should only be enabled once the PR is ready to go and can only be enabled by a maintainer

**Review by leerob — APPROVED** [review]

**Comment by michaelangeloio** [comment]
@leerob needs workflow approval after merging canary

**Review by leerob — APPROVED** [review]

**Pull request merged**
